### PR TITLE
Rest controllers + implement service logic

### PR DESCRIPTION
**Pull request by mrnchernova**

### CODE REVIEW

File: src/main/java/by/project/trucking_v2/controller/CompletedOrderController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @Autowired
+    private CompletedOrderService completedOrderService;
+
+    @GetMapping
+    public List<CompletedOrder> getAll() {
+        return completedOrderService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public CompletedOrder getById(@PathVariable("id") int id) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/CompletedOrderController.java]: Integer

File: src/main/java/by/project/trucking_v2/controller/DriverController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        return driverService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public Driver getById(@PathVariable("id") int id) {
+        return driverService.findById(id);
+    }
+
+    @PostMapping
+    public Driver create(@RequestBody Driver driver, @RequestParam int le_id) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/DriverController.java]: le_id? what's that?

Comment (RomanKashenok) [reply src/main/java/by/project/trucking_v2/controller/DriverController.java]: add it to body. in body paste driver and legal entity id (or whole legal entity)

File: src/main/java/by/project/trucking_v2/controller/LegalEntityController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        return legalEntityService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public LegalEntity getById(@PathVariable("id") int id) {
+        return legalEntityService.findById(id);
+    }
+
+    @PutMapping("/{id}")
+    public LegalEntity update(@PathVariable("id") int id, @RequestBody LegalEntity legalEntity) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/LegalEntityController.java]: remove pah variable

File: src/main/java/by/project/trucking_v2/controller/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        return orderService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public Order getById(@PathVariable("id") int id) {
+        return orderService.findById(id);
+    }
+
+    @PostMapping
+    public Order create(@RequestBody Order order, @RequestParam int le_id) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/OrderController.java]: same as with Driver

File: src/main/java/by/project/trucking_v2/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        return orderService.save(order, le_id);
+    }
+
+    @PutMapping("/{id}")
+    public Order update(@PathVariable("id") int id, @RequestBody Order order) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/OrderController.java]: same as with Driver

File: src/main/java/by/project/trucking_v2/controller/OrderController.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+
+
+    /**
+     * Обрабатывает событие, когда Перевозчик выберет заказ
+     * Статус заказа изменится на IN_PROGRESS
+     * Сам Перевозчик будет сохранен в сессии (скоро, но пока не работает)
+     */
+    @PutMapping("/deal/{id}")
+    public Order orderChoice(@PathVariable("id") int id, @RequestBody Order order) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/OrderController.java]: the same

File: src/main/java/by/project/trucking_v2/controller/TransportController.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        return transportService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public Transport getById(@PathVariable("id") int id) {
+        return transportService.findById(id);
+    }
+
+    @PostMapping
+    public Transport create(@RequestBody Transport transport, @RequestParam int le_id) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/TransportController.java]: same

File: src/main/java/by/project/trucking_v2/controller/TransportController.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        return transportService.save(transport, le_id);
+    }
+
+    @PutMapping("/{id}")
+    public Transport update(@PathVariable("id") int id, @RequestBody Transport transport) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/TransportController.java]: same

File: src/main/java/by/project/trucking_v2/controller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    public User getUserById(@PathVariable("id") int id) {
+        return userService.findById(id);
+    }
+
+    @PostMapping
+    public User create(@RequestBody User user) {
+        return userService.save(user);
+    }
+    @PutMapping("/{id}")
+    public User updateUser(@PathVariable("id") int id, @RequestBody User user) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/UserController.java]: same

File: src/main/java/by/project/trucking_v2/exception/NotFoundException.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    public static final String ANSI_BLUE = "\u001B[34m";
+    public static final String ANSI_PURPLE = "\u001B[35m";
+    public static final String ANSI_CYAN = "\u001B[36m";
+    public static final String ANSI_WHITE = "\u001B[37m";
+    public NotFoundException() {
+        super();
+
+        final Logger logger = Logger.getLogger(NotFoundException.class.getName());
+        logger.warning("Мы не знаем что это такое, если бы мы знали что это такое, мы не знаем что это такое");
+        logger.severe("404. Such id not found");
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/exception/NotFoundException.java]: remove all logic from constructor

File: src/main/java/by/project/trucking_v2/repository/UserRepository.java [hunk 11]
```diff
@@ -2,9 +2,12 @@
 
 import by.project.trucking_v2.model.User;
 import org.springframework.data.repository.CrudRepository;
-import org.springframework.stereotype.Repository;
 
-@Repository
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/repository/UserRepository.java:7]: why do you removed it?

File: src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class CompletedOrderServiceImpl implements CompletedOrderService {
+    public static final String ANSI_RED = "\u001B[31m";
+    public static final String ANSI_YELLOW = "\u001B[33m";
+    public static final String ANSI_GREEN = "\u001B[32m";
+    public static final String ANSI_RESET = "\u001B[0m";
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java]: let's not use such beautification.. just forget about it

File: src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    private static final Logger log = LoggerFactory.getLogger(OrderServiceImpl.class);
+
+    @Autowired
+    CompletedOrderRepository completedOrderRepository;
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java]: why not private as others?

File: src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+
+    /**
+     * Вызывается после выбора Заказа Перевозчиком
+     * Перевозчик сам определяет какого Водителя и какую Машину назначить для выполнения заказа
+     * следовательно передает driver_id и transport_id
+     */
+    @Override
+    public CompletedOrder orderInProgress(int id_order) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java]: orderId

File: src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+        log.info(ANSI_GREEN + "находим нужный заказ" + ANSI_RESET);
+        Order currentOrder = orderRepository.findById(id_order).orElseThrow(NotFoundException::new);
+
+        /*это будет передаваться* через форму*/
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/service/CompletedOrderServiceImpl.java:54]: ничего не понятно, но очень интересно )) завтра расскажешь

File: src/main/java/by/project/trucking_v2/service/OrderServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+    public Order save(Order order, int le_id) {
+        LegalEntity le = legalEntityRepository.findById(le_id).orElseThrow(NotFoundException::new);
+        Order newOrder = new Order();
+        newOrder.setDescription(order.getDescription());
+        newOrder.setWeight(order.getWeight());
+        newOrder.setDeparture(order.getDeparture());
+        newOrder.setDestination(order.getDestination());
+        newOrder.setPrice(order.getPrice());
+        newOrder.setStatus(order.getStatus());
+        newOrder.setLegalEntity(le);
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/service/OrderServiceImpl.java]: and how differs order and newOrder? Maybe let's save just 'order'

File: src/main/java/by/project/trucking_v2/service/TransportServiceImpl.java [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+        return transports;
+    }
+
+    @Override
+    public Transport findById(int id) {
+        return transportRepository.findById(id).orElseThrow(() -> new NotFoundException());
+    }
+
+    @Override
+    public Transport save(Transport transport, int le_id) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/service/TransportServiceImpl.java]: same as with order

**Review by RomanKashenok — CHANGES_REQUESTED** [review]
Also mark every service method, that update DB state as @Transactional. And read a lil bit about it

File: src/main/java/by/project/trucking_v2/repository/UserRepository.java (hunk 11 above)
Comment (mrnchernova) [reply src/main/java/by/project/trucking_v2/repository/UserRepository.java:7]: случайно

File: src/main/java/by/project/trucking_v2/controller/LegalEntityController.java (hunk 3 above)
Comment (mrnchernova) [reply src/main/java/by/project/trucking_v2/controller/LegalEntityController.java]: А как мы тогда сможем менять LegalEntity?

File: src/main/java/by/project/trucking_v2/controller/DriverController.java [hunk 18]
```diff
@@ -19,22 +19,22 @@ public List<Driver> getAll() {
...
     @PostMapping
-    public Driver create(@RequestBody Driver driver, @RequestParam int le_id) {
-        return driverService.save(driver, le_id);
+    public Driver create(@RequestBody Driver driver) {
+        return driverService.save(driver);
     }
 
     @PutMapping("/{id}")
-    public Driver update(@PathVariable("id") int id, @RequestBody Driver driver) {
+    public Driver update(@PathVariable Integer id, @RequestBody Driver driver) {
```
Comment (RomanKashenok) [inline src/main/java/by/project/trucking_v2/controller/DriverController.java:32]: Ок. Но я не понимаю зачем тут айдишник отдельно

**Review by RomanKashenok — APPROVED** [review]

**Pull request merged**